Pd 1057 new content for expanding a raidz storage pool (#2931)
truenas/documentation: content/References/ExtensionCalculator.md

---
title: "RAIDZ Extension Calculator"
description: "Calculator to determine usable ZFS capacity and other ZFS pool metrics, and to compare storage pool layouts."
weight: 90
tags:
- zfs
- storage
- raidz
---

RAIDZ extension allows TrueNAS users to add disks one at a time to an existing RAIDZ VDEV, incrementally expanding its capacity.
Use this calculator to estimate the storage capacity efficiency of a ZFS RAIDZ VDEV after incremental extension.

Existing data blocks retain their original data-to-parity ratio and block width, but spread across the larger set of disks.
New data blocks adopt the new data-to-parity ratio and width.
Because of this overhead, an extended RAIDZ VDEV can report a lower total capacity than a freshly created VDEV with the same number of disks.

> Excess overhead, or lost headroom, can be recovered by replicating and rewriting existing data to the extended pool.

Author: ```suggestion

Extended VDEVs recover lost headroom as existing data is read and rewritten to the new parity ratio.
This occurs naturally over the lifetime of the pool as you modify or delete data.
To manually recover capacity, simply replicate and rewrite the data to the extended pool.
```

See [Extending a RAIDZ VDEV]({{< relref "ManagePoolsSCALE.md #extending-a-raidz-vdev" >}}) for more information, including a detailed overview, considerations, and a tutorial.

{{< expand "How to use this calculator" "v" >}}

> Edit the values to see how much lost headroom capacity could be gained by rewriting existing data after extension.

Author: ```suggestion
Edit the values to see how much lost headroom capacity is available to recover.
```

{{< truetable >}}
| Value | Description |
|-----------|-------------|
| **Drive Size** | Enter the size (in TB) of each disk in the existing VDEV. Added disks should be the same size or larger than existing drives. |
| **RAIDZ Level / Number of Parity Disks** | Enter the RAIDZ level (RAIDZ1, RAIDZ2, or RAIDZ3) of the existing VDEV. Extending a VDEV does not change RAIDZ fault tolerance. |
| **Data expansion threshold** | Enter the percentage of pre-extension VDEV capacity that is filled before adding an additional disk. |
| **Initial number of drives** | Enter the width of the VDEV before extension. |
| **Final number of drives** | Enter the planned final width of the VDEV after extension. |
{{< /truetable >}}
{{< /expand >}}

{{< expand "How to interpret the results" "v" >}}

> Calculated values represent how much space on the final extended VDEV is lost to headroom, or data capacity that would be available in an equivalent VDEV created from scratch with the same width and data-to-parity ratio.

Author: Since this is a hypothetical for a VDEV that doesn't and will not actually exist, I think active present tense here detracts from the meaning
